### ISSUE TITLE
clippy suggests`40..to_radians()` which is a compile error

### ISSUE THREAD

**Issue by Volker-Weissmann** [rust-lang/rust-clippy#7651]
```rust
use std::f64::consts::PI;
fn main() {
    40. * PI / 180.;
}
```
`cargo clippy -- -W clippy::nursery`
```
warning: conversion to radians can be done more accurately
 --> src/main.rs:3:5
  |
3 |     40. * PI / 180.;
  |     ^^^^^^^^^^^^^^^ help: consider using: `40..to_radians()`
  |
  = note: `-W clippy::suboptimal-flops` implied by `-W clippy::nursery`
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#suboptimal_flops
```
Clippy suggests `40..to_radians()`, which is a compile-error. `40_f64.to_radians()` would be correct.

Slightly OT: This bug seems to be catchable by a fuzzer that verifies that clippy never suggest introducing compiler errors into code that compiles.

### Meta

**Rust version (`rustc -Vv`):**

```
rustc 1.55.0-nightly (ce1d5611a 2021-06-18)
binary: rustc
commit-hash: ce1d5611a28468663e275078649e7ca6eef735a8
commit-date: 2021-06-18
host: x86_64-unknown-linux-gnu
release: 1.55.0-nightly
LLVM version: 12.0.1
```

@rustbot label +suggestion-causes-error

**Comment by Dmitry-Borodin** [comment]
Reproduced it in current latest stable toolchain (rustc 1.55.0)
I will try to solve it in a next few days.
@rustbot claim
